### PR TITLE
Simplify and fix macho trie checks

### PR DESCRIPTION
**Pull request by thestr4ng3r**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Fix regression in fuzz tests by e2edd9d1ef79842e772e21f07b4992438614083d

**Test plan**

Build with clang+optimization on linux, then
valgrind rizin test/bins/fuzzed/65940f6c970bb373444e0d0aab817edc
to detect the original issues.

### CODE REVIEW

File: librz/bin/format/mach0/mach0.c [hunk 1]
```diff
@@ -2751,12 +2752,12 @@ static int walk_exports(struct MACH0_(obj_t) * bin, RExportsIterator iterator, v
 			RZ_FREE(next);
 			goto beach;
 		}
-		if (SZT_ADD_OVFCHK((size_t)tr, (size_t)trie) || (size_t)tr + trie >= end) {
+		if (tr >= (ut64)(end - trie)) {
```
Comment (ret2libc) [inline librz/bin/format/mach0/mach0.c]: why not the OVFCHK check?

File: librz/bin/format/mach0/mach0.c [hunk 2]
```diff
@@ -2751,12 +2752,12 @@ static int walk_exports(struct MACH0_(obj_t) * bin, RExportsIterator iterator, v
...
 			RZ_LOG_ERROR("malformed export trie\n");
 			RZ_FREE(next);
 			goto beach;
 		}
-		next->node = (size_t)tr + trie;
+		next->node = trie += (size_t)tr;
```
Comment (ret2libc) [inline librz/bin/format/mach0/mach0.c]: it's probably me, but i feel this is more obscure to read.

Comment (thestr4ng3r) [reply librz/bin/format/mach0/mach0.c]: My change was not only obscure, but also wrong :-)
Fixed now.

File: librz/bin/format/mach0/mach0.c (hunk 1 above)
Comment (thestr4ng3r) [reply librz/bin/format/mach0/mach0.c]: Simplified it even more. Explicit overflow-checking is just not necessary anymore now.

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Pull request merged**